staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -199,11 +199,18 @@ var longRunningFilter = genericfilters.BasicLongRunningRequestCheck(sets.NewStri
 
 func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	ctx := req.Context()
-	requestInfo, ok := apirequest.RequestInfoFrom(ctx)
-	if !ok {
-		responsewriters.InternalError(w, req, fmt.Errorf("no RequestInfo found in the context"))
-		return
+	req, _ = apifilters.SetShortPathRequestTenant(req)
+	requestInfo, _ := apirequest.RequestInfoFrom(ctx)
+
+	crdTenant := requestInfo.Tenant
+	// SetShortPathRequestTenant does not update the tenant in a short-path request if it is a system-user request,
+	// as it may be an operation across all tenants. Also in some integration tests, the tenant in the requestInfo is not set.
+	// However, if code hits here, we know it is a request across multiple tenants,
+	// so we can set the tenant here.
+	if crdTenant == metav1.TenantNone {
+		crdTenant = metav1.TenantDefault
 	}
```

Comment:
Please fix the problem of missing "not" in the comment, otherwise it's really confusing :) Also should we use meta1.TenantSystem here instead of meta1.TenantDefault?

Reply:
Updated the comments. 

We need to use TenantDefault here for now. It will be updated to TenantSystem in the PR that we move everything under TenantDefault to TenantSystem.